crates/sncast/src/starknet_commands/deploy.rs

```diff
@@ -20,49 +23,120 @@ use starknet::signers::LocalWallet;
 #[command(about = "Deploy a contract on Starknet")]
 pub struct Deploy {
     /// Class hash of contract to deploy
-    #[clap(short = 'g', long)]
-    pub class_hash: FieldElement,
+    #[clap(short = 'g', long, conflicts_with = "contract_name")]
+    pub class_hash: Option<FieldElement>,
 
-    /// Calldata for the contract constructor
-    #[clap(short, long, value_delimiter = ' ', num_args = 1..)]
-    pub constructor_calldata: Vec<FieldElement>,
+    // Name of the contract to deploy
+    #[clap(long, conflicts_with = "class_hash")]
+    pub contract_name: Option<String>,
 
-    /// Salt for the address
-    #[clap(short, long)]
-    pub salt: Option<FieldElement>,
-
-    /// If true, salt will be modified with an account address
-    #[clap(long)]
-    pub unique: bool,
+    #[clap(flatten)]
+    pub args: DeployArgs,
 
     #[clap(flatten)]
     pub fee_args: FeeArgs,
 
-    /// Nonce of the transaction. If not provided, nonce will be set automatically
-    #[clap(short, long)]
-    pub nonce: Option<FieldElement>,
-
-    /// Version of the deployment (can be inferred from fee token)
-    #[clap(short, long)]
-    pub version: Option<DeployVersion>,
-
     #[clap(flatten)]
     pub rpc: RpcArgs,
 }
 
-#[derive(ValueEnum, Debug, Clone)]
-pub enum DeployVersion {
-    V1,
-    V3,
+impl From<DeclareDeploy> for Deploy {
+    fn from(declare_deploy: DeclareDeploy) -> Self {
+        let DeclareDeploy {
+            contract_name,
+            deploy_args,
+            fee_token,
+            rpc,
+        } = declare_deploy;
+
+        let fee_args = FeeArgs {
+            fee_token: Some(fee_token),
+            ..Default::default()
+        };
+
+        Deploy {
+            class_hash: None,
+            contract_name: Some(contract_name),
+            args: deploy_args,
+            fee_args,
+            rpc,
+        }
+    }
+}
+
+impl Deploy {
+    pub fn build_artifacts_and_get_compiled_contract(
+        &self,
+        json: bool,
+        profile: &Option<String>,
+    ) -> Result<CompiledContract> {
+        let contract_name = self
+            .contract_name
+            .clone()
+            .ok_or(anyhow!("Contract name and class hash unspecified"))?;
```

Comment:
I think we could use `ok_or_else` here for super minor optimisation. 

```suggestion
            .ok_or_else(|| anyhow!("Contract name and class hash unspecified"))?;
```

Reply:
And in other places as well